python.d/tomcat.chart.py

```diff
@@ -16,61 +18,141 @@
+        'options': [None, "Threads", "current threads", "statistics", "tomcat.threads", "area"],
         'lines': [
             ["currentThreadCount", 'current', "absolute"],
             ["currentThreadsBusy", 'busy', "absolute"]
         ]},
     'jvm': {
-        'options': [None, "JVM Free Memory", "MB", "statistics", "tomcat.jvm", "area"],
+        'options': [None, "JVM Memory Pool Usage", "MB", "memory", "tomcat.jvm", "stacked"],
         'lines': [
-            ["free", None, "absolute", 1, 1048576]
-        ]}
+            ["free",  'free', "absolute", 1, 1048576],
+            ["eden_used", 'eden', "absolute", 1, 1048576],
+            ["survivor_used", 'survivor', "absolute", 1, 1048576],
+            ["tenured_used", 'tenured', "absolute", 1, 1048576],
+            ["code_cache_used", 'code cache', "absolute", 1, 1048576],
+            ["compressed_used", 'compressed', "absolute", 1, 1048576],
+            ["metaspace_used", 'metaspace', "absolute", 1, 1048576],
+        ]},
+    'jvm_eden': {
+        'options': [None, "Eden Memory Usage", "MB", "memory", "tomcat.jvm_eden", "area"],
+        'lines': [
+            ["eden_used", 'used', "absolute", 1, 1048576],
+            ["eden_commited", 'commited', "absolute", 1, 1048576],
+            ["eden_max", 'max', "absolute", 1, 1048576]
+        ]},
+    'jvm_survivor': {
+        'options': [None, "Survivor Memory Usage", "MB", "memory", "tomcat.jvm_survivor", "area"],
+        'lines': [
+            ["survivor_used", 'used', "absolute", 1, 1048576],
+            ["survivor_commited", 'commited', "absolute", 1, 1048576],
+            ["survivor_max", 'max', "absolute", 1, 1048576]
+        ]},
+    'jvm_tenured': {
+        'options': [None, "Tenured Memory Usage", "MB", "memory", "tomcat.jvm_tenured", "area"],
+        'lines': [
+            ["tenured_used", 'used', "absolute", 1, 1048576],
+            ["tenured_commited", 'commited', "absolute", 1, 1048576],
+            ["tenured_max", 'max', "absolute", 1, 1048576]
+        ]},
 }
 
-
 class Service(UrlService):
     def __init__(self, configuration=None, name=None):
         UrlService.__init__(self, configuration=configuration, name=name)
         self.url = self.configuration.get('url', "http://127.0.0.1:8080/manager/status?XML=true")
+        self.connector_name = self.configuration.get('connector_name', None)
         self.order = ORDER
         self.definitions = CHARTS
 
     def check(self):
-        netloc = urlparse(self.url).netloc.rpartition(':')
-        if netloc[1] == ':': port = netloc[2]
-        else: port = 80
-        
-        self.regex_jvm = compile(r'<jvm>.*?</jvm>')
-        self.regex_connector = compile(r'[a-z-]+%s.*?/connector' % port)
-        self.regex = compile(r'([\w]+)=\\?[\'\"](\d+)\\?[\'\"]')
-        
-        return UrlService.check(self)
+        return UrlService.check(self) and self._get_data() != None
```

Comment:
check method not needed (looks like the [UrlService check](https://github.com/firehol/netdata/blob/master/python.d/python_modules/base.py#L568-L591) is ok for this plugin)